base/collection_n1ql.go

```diff
@@ -119,8 +121,50 @@ func (c *Collection) CreatePrimaryIndex(indexName string, options *N1qlIndexOpti
 	return CreatePrimaryIndex(c, indexName, options)
 }
 
-func (c *Collection) WaitForIndexOnline(indexName string) error {
-	return WaitForIndexOnline(c, indexName)
+// WaitForIndexOnline takes set of indexes and watches them till they're online.
+func (c *Collection) WaitForIndexesOnline(indexNames []string, watchPrimary bool) error {
+	logCtx := context.TODO()
+	mgr := c.cluster.QueryIndexes()
+	maxNumAttempts := 250
+	retrySleeper := CreateMaxDoublingSleeperFunc(maxNumAttempts, 100, 5000)
+	retryCount := 0
+	var waitErr error
+
+	watchOption := gocb.WatchQueryIndexOptions{
+		WatchPrimary:   watchPrimary,
+		RetryStrategy:  &goCBv2FailFastRetryStrategy{},
+		ScopeName:      c.ScopeName(),
+		CollectionName: c.Name(),
+	}
+	indexOption := gocb.GetAllQueryIndexesOptions{
+		ScopeName:      c.ScopeName(),
+		CollectionName: c.Name(),
+	}
+
+	for {
+		waitErr = mgr.WatchIndexes(c.BucketName(), indexNames, waitTime, &watchOption)
```

Comment:
I'd just drop this and check each index in `GetAllIndexes` is online

Reply:
I think we need this here as this is how we ensure that the indexes we specify/need are there and online where as the get all indexes is purely for the logging purposes. For example, i tried doing this changed the validate index code in the index_test.go to point to this function and tests failed due to a bucket having non xattr indexes when it was expecting them.

Reply:
I was suggesting basically keeping a list of onlineIndexes, and once onlineIndexes == indexNames we'll break out of this for loop

Reply:
Updated the PR with this approach.